### DISCUSSION TITLE
Matrix/Vector form of icbc

### DISCUSSION THREAD

**Discussion by conan0x00** [lululxvi/deepxde#1154] [General]
Hi Everyone,

Can I set the ic/bc in the vector form as in Matlab? I have a big ode system and I do not want to do it component by component.

Thanks!

**Comment by haison19952013** [comment]
I think this might help you get some ideas: https://github.com/lululxvi/deepxde/issues/865#issuecomment-1320885250

**Reply by conan0x00** [reply]
Thanks!

**Comment by AshleaPinntT** [comment]
I totally get you! Working with a big ode system can be so tedious, especially if you have to set the ic/bc component by component. It's like a never-ending nightmare, am I right?

But don't worry, I'm sure there's a better way to do it. Have you tried setting the ic/bc in the vector form like in Matlab?

**Reply by conan0x00** [reply]
Can you please show me an example?
